clippy_lints/src/lib.rs

```diff
@@ -466,6 +467,23 @@ pub fn read_conf(sess: &Session) -> Conf {
         .emit();
     }
 
+    if conf.msrv.is_none() {
+        // let's try to get msrv from `Cargo.toml`s field `rust-version`
+        match MetadataCommand::new().no_deps().exec() {
+            Ok(metadata) => {
+                conf.msrv = metadata
+                    .packages
+                    .get(0)
+                    .and_then(|x| x.rust_version.as_ref())
+                    .and_then(|r| r.comparators.get(0))
+                    .map(|v| format!("{}.{}.{}", v.major, v.minor.unwrap_or(0), v.patch.unwrap_or(0)));
+            },
+            Err(e) => {
+                sess.struct_err(&format!("could not read cargo metadata: {}", e)).emit();
```

Comment:
I don't think this is a hard error. For example with `clippy-driver` outside of a `cargo` project, no `Cargo.toml` file can be found. I would just silently keep `conf.msrv = None` here.